apps/frontend/src/libs/components/search/libs/hooks/use-search.hook.ts

```diff
@@ -0,0 +1,30 @@
+import { QueryParameterName } from "~/libs/enums/enums.js";
+import { useEffect, useSearchParams, useState } from "~/libs/hooks/hooks.js";
+
+type UseSearch = {
+	search: string;
+	setSearch: (search: string) => void;
+};
+
+const useSearch = (): UseSearch => {
```

Comment:
we need to rename setSearch to onSearch before exposing it
```suggestion
const useSearch = (): {
	search: string;
	onSearch: (search: string) => void;
} => {
```
```tsx
	return {
		search,
		onSearch: setSearch,
	};
```